### ISSUE TITLE
[flow] "Could not decide which case to select" when passing functional components to Field's component

### ISSUE THREAD

**Issue by abmai** [erikras/redux-form#3878]
### Are you submitting a **bug report** or a **feature request**?

Bug

### What is the current behavior?

When passing Functional Components to a <Field>'s `component` prop, Flow is barfing because it can't decide between a Function type or a React.ComponentType type.

### What is the expected behavior?
No errors :/

### What's your environment?

Redux-form: 7.2.3

### Other information

```
162:       name={fieldName}
                      ^^^^^^^^^^^^^^^^^^^^^^^ function. Could not decide which case to select
  8:   component: React.ComponentType<*> | Function | string,
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ union type. See: node_modules/redux-form/lib/FieldProps.types.js.flow:8
  Case 1 may work:
    8:   component: React.ComponentType<*> | Function | string,
                    ^^^^^^^^^^^^^^^^^^^^^^ ComponentType. See: node_modules/redux-form/lib/FieldProps.types.js.flow:8
  But if it doesn't, case 2 looks promising too:
    8:   component: React.ComponentType<*> | Function | string,
                                             ^^^^^^^^ function type. See: node_modules/redux-form/lib/FieldProps.types.js.flow:8
  Please provide additional annotation(s) to determine whether case 1 works (or consider merging it with case 2):
  139:   return (
                 ^ return
```

**Comment by xiankai** [comment]
For the time being what I'm doing is to cast the functional component:

```
import type { StatelessFunctionalComponent } from 'react';

const functionalComponent: StatelessFunctionalComponent<*> = props => jsx;
```

**Comment by asazernik** [comment]
Can confirm this is still an issue in 8.1.0

**Comment by asazernik** [comment]
The problem is that the type annotation is redundant; a function `FieldProps => React.Node` **is** a `Component<FieldProps>`. Creating a PR.

**Comment by erikras** [comment]
Published fix in [`v8.2.0`](https://github.com/erikras/redux-form/releases/tag/v8.2.0).
